R4Ghidra/src/main/java/r4ghidra/repl/R2REPLImpl.java

```diff
@@ -548,6 +553,91 @@ private String executeQuotedCommand(String quotedStr) throws R2CommandException
     return executeCommandWithHandler(cmd);
   }
 
+  /**
+   * Execute a command that starts with a brace '{'. This implements r2pipe's JSON
+   * braced command behavior. The expected JSON is: {"cmd":"...","json":false,"trim":true}
+   * If the input is "{?" the usage string is returned.
+   */
+  private String executeBracedCommand(String bracedStr) throws R2CommandException {
+    // Quick usage help: if the second char is '?' return usage (mimic r2pipe.c behavior)
+    if (bracedStr.length() > 1 && bracedStr.charAt(1) == '?') {
+      return "Usage: {\"cmd\":\"...\",\"json\":false,\"trim\":true} # `cmd` is required\n";
+    }
+
+    // Try to extract the JSON text. The input usually starts with '{' and may end with '}'.
+    String raw = bracedStr.trim();
+    String jsonText = raw;
+    // If there are extra characters after the last closing brace, trim them off
+    int lastBrace = raw.lastIndexOf('}');
+    if (lastBrace != -1) {
+      jsonText = raw.substring(0, lastBrace + 1);
+    }
+
+    // Parse JSON
+    org.json.JSONObject out = new org.json.JSONObject();
+    try {
+      org.json.JSONObject j = new org.json.JSONObject(jsonText);
+
+      // cmd is required
+      if (!j.has("cmd")) {
+        out.put("res", "");
+        out.put("error", true);
+        out.put("value", 0);
+        out.put("code", context.getLastErrorCode());
+        return out.toString() + "\n";
+      }
+
+      String r_cmd = j.getString("cmd");
+      boolean is_json = j.optBoolean("json", false);
+      boolean is_trim = j.has("trim") ? j.optBoolean("trim", true) : true; // default true
+
+      // Execute the command
+      String res = executeCommand(r_cmd);
```

Comment:
If the r2 command fails we still set `"error":false` in the response JSON. Is this the expected behavior?

Reply:
probably not, the response should always have all those fields. i may probably write a json schema 

{"res":"he\n","error":false,"value":256,"code":0,"code":0}